### PR TITLE
Compiler warning part 5

### PR DESCRIPTION
**Pull request by RincewindsHat**
Parts of #1848
Easier to digest :-)

### CODE REVIEW

File: plugins/check_snmp.c [hunk 1]
```diff
@@ -1185,7 +1187,7 @@ multiply (char *str)
 	if(verbose>2)
 		printf("    multiply extracted double: %f\n", val);
 	val *= multiplier;
-	if (fmtstr != "") {
+	if (strcmp(fmtstr, "") != 0) {
```
[the remark this answers is not in the document]
Comment (RincewindsHat) [reply plugins/check_snmp.c:1190]: would you define an empty format string as a valid use case?

[the remark this answers is not in the document]
Comment (RincewindsHat) [reply plugins/check_snmp.c:1190]: but yes, in principle you are, of course, right.

[the remark this answers is not in the document]
Comment (RincewindsHat) [reply plugins/check_snmp.c]: this should work as intented now, I think

**Comment by RincewindsHat** [comment]
@andreasbaumann @sni can I have a review for this? By now it is practically comprehensible.

**Review by sni — APPROVED** [review]
looks good, how did that work before anyway? :)

**Pull request merged**